boot/freeldr/freeldr/arch/i386/pc98/machpc98.c

```diff
@@ -0,0 +1,162 @@
+/*
+ * PROJECT:     FreeLoader
+ * LICENSE:     GPL-2.0-or-later (https://spdx.org/licenses/GPL-2.0-or-later)
+ * PURPOSE:     Hardware-specific routines for NEC PC-98 series
+ * COPYRIGHT:   Copyright 2020 Dmitry Borisov (di.sean@protonmail.com)
+ */
+
+/* INCLUDES *******************************************************************/
+
+#include <freeldr.h>
+
+#include <debug.h>
+DBG_DEFAULT_CHANNEL(HWDETECT);
+
+/* GLOBALS ********************************************************************/
+
+BOOLEAN HiResoMachine;
+
+/* FUNCTIONS ******************************************************************/
+
+VOID
+Pc98GetExtendedBIOSData(PULONG ExtendedBIOSDataArea, PULONG ExtendedBIOSDataSize)
+{
+    *ExtendedBIOSDataArea = HiResoMachine ? MEM_EXTENDED_HIGH_RESO : MEM_EXTENDED_NORMAL;
+    *ExtendedBIOSDataSize = 64;
+}
+
+VOID
+Pc98HwIdle(VOID)
+{
+    /* Unimplemented */
+}
+
+VOID
+Pc98PrepareForReactOS(VOID)
+{
+    Pc98DiskPrepareForReactOS();
+    Pc98VideoPrepareForReactOS();
+    DiskStopFloppyMotor();
+}
+
+ULONG
+Pc98GetBootSectorLoadAddress(IN UCHAR DriveNumber)
```

Comment:
I think then that your fat12pc98 image should be instead based on some different sector address?
https://git.reactos.org/?p=reactos.git;a=blob;f=boot/freeldr/bootsect/CMakeLists.txt;hb=74eda39118d9a266a92ba79ca033e39f1f6b5005#l12
Says "7c00" but may be instead 1fc0 ?

Reply:
Good question! Indeed, the base address should be set to `0`. However, it's not easy to do with current `fathelp.inc`. Value `0x7C00' is hardcoded. I use [code relocation](https://git.reactos.org/?p=reactos.git;a=blob;f=boot/freeldr/bootsect/pc98/fat12fdd.S;hb=74eda39118d9a266a92ba79ca033e39f1f6b5005#l137) for compatibility purpose.

Reply:
Then this relocation trick should work, as long as the PC-98 bios does not store any other "vital" information at (or around) 7C00.

Reply:
`7C00` value is inside conventional memory area.

Reply:
Then this should be fine :)